### ISSUE TITLE
Update benchmark suite for v1.1.0 sample_token API compatibility

### ISSUE THREAD

**Issue by GarthDB** [GarthDB/metal-candle#35]
## Description

The `inference` and `training` benchmarks need updates to be compatible with the v1.1.0 `sample_token` API changes.

## Context

- v1.1.0 introduced breaking changes to `sample_token` signature
- Benchmarks still use v1.0.0 API (missing repetition penalty parameters)
- Does not affect v1.2.0 fused softmax integration (already validated in PR #27)

## Breaking Change (v1.1.0)

**Before (v1.0.0):**
```rust
pub fn sample_token(logits: &Tensor, strategy: &SamplingStrategy) -> Result<u32>
```

**After (v1.1.0):**
```rust
pub fn sample_token(
    logits: &Tensor, 
    strategy: &SamplingStrategy,
    generated_ids: &[u32],      // NEW: Previously generated tokens
    repetition_penalty: f32,     // NEW: Penalty factor (1.0 = no penalty)
) -> Result<u32>
```

## Affected Benchmarks

- `benches/inference.rs`
- `benches/training.rs`

## Migration Required

Update `sample_token` calls:
```rust
// Old (v1.0.0)
let token = sample_token(&logits, &strategy)?;

// New (v1.1.0) - no repetition penalty
let token = sample_token(&logits, &strategy, &[], 1.0)?;
```

## Acceptance Criteria

- [ ] `inference.rs` benchmark compiles and runs with v1.1.0 API
- [ ] `training.rs` benchmark compiles and runs with v1.1.0 API
- [ ] Benchmark results remain comparable to previous runs
- [ ] All benchmarks pass in CI smoke test workflow

## References

- CHANGELOG v1.1.0 Breaking Changes section
- CHANGELOG v1.2.0 Known Issues section
- PR #33 implementation

## Priority

**Medium** - Should be fixed in v1.2.1 for complete benchmark suite coverage
